### PR TITLE
Update lit_gpt commit to PyTorch 2.2

### PR DESCRIPTION
**Pull request by Chris113113**
The PR's primary purpose is updating lit-gpt's commit to a PyTorch 2.2 commit. This also comes with a few other things:

Logs from new image: http://shortn/_klGl5LKuQm

* Removes building of `flash_attn` resulting in more reliable lit-gpt image builds.
* Removes Lightning.Trainer in favor of Lightning.Fabric (provided by the lit-gpt openwebtext sample)
* Refactors most of the nsight workflow into a separate Callback class.
* Updates the base image to 24.01, which comes bundled with PyTorch 2.2, and NCCL 2.19.
* Updates the ncclPlugin and rxdmContainer versions
* Adds volume mount for /data onto the host which prevents re-downloading the dataset on subsequent runs.

### CODE REVIEW

**Review by gkroiz — COMMENTED** [review]
Mostly LGTM. In addition to my comments, two things:
1) From this PR it looks like we are adjusting this example to run Llama-2-13B instead of Llama-2-70B. Just want to double check this is intentional
2) Could you attach some link (using short-gen) where you are able to run this workload?

File: sample_workloads/lit-gpt-demo/build_and_push_litgpt.sh [hunk 1]
```diff
@@ -13,11 +13,11 @@ FULL_IMAGE=${FULL_IMAGE:="$ARTIFACT_REGISTRY/litgpt-full"}
 # Clone LitGPT and checkout a flash-attn enabled commit
 if [ ! -d $LITGPT_PATH ]; then
     git clone https://github.com/Lightning-AI/lit-gpt.git
-    cd lit-gpt
-    git checkout d5d371417ecb3d3b6c4f30837d8bb7cf2b5310ae
-    cd ..
     LITGPT_PATH=lit-gpt
 fi
+cd lit-gpt
+git checkout 44c3c58b759fa0903ab31ed8863a66c157d5ccd9
```
[the remark this answers is not in the document]
Comment (Chris113113) [reply sample_workloads/lit-gpt-demo/build_and_push_litgpt.sh:19]: No hard reason, it was HEAD when I started to experiment with 2.2.

File: sample_workloads/lit-gpt-demo/sample.env [hunk 2]
```diff
@@ -0,0 +1,15 @@
+MODEL_NAME=Llama-2-13b-hf
+WARMUP_ITERS=10
+EXPERIMENT_ROOT_DIR=llama2-13b
+NCCL_DEBUG=TRACE
```
[the remark this answers is not in the document]
Comment (Chris113113) [reply sample_workloads/lit-gpt-demo/sample.env:4]: Deleted this file.

**Comment by Chris113113** [comment]
> Mostly LGTM. In addition to my comments, two things:
> 
> 1. From this PR it looks like we are adjusting this example to run Llama-2-13B instead of Llama-2-70B. Just want to double check this is intentional
> 2. Could you attach some link (using short-gen) where you are able to run this workload?

1. Good catch on 13B, I was using it to experiment.

2. https://pastebin.com/1iKLXCMf, added to description.

File: sample_workloads/lit-gpt-demo/helm/values.yaml [hunk 3]
```diff
@@ -1,22 +1,25 @@
 cluster:
-  nNodes: 8
+  nNodes: 4
```
Comment (gkroiz) [inline sample_workloads/lit-gpt-demo/helm/values.yaml:2]: want to double check here, do we expect this value to be 4?

**Review by gkroiz — APPROVED** [review]
I added one more small comment, otherwise LGTM

File: sample_workloads/lit-gpt-demo/helm/values.yaml (hunk 3 above)
Comment (Chris113113) [reply sample_workloads/lit-gpt-demo/helm/values.yaml:2]: Yep, defaulting to 4.

**Pull request merged**